modules/representation_management/app/models/representation_management/form_2122_data.rb

```diff
@@ -2,8 +2,26 @@
 
 module RepresentationManagement
   class Form2122Data < RepresentationManagement::Form2122Base
-    attr_accessor :organization_name
+    def organization_name
+      find_organization_name
+    end
 
-    validates :organization_name, presence: true
+    attr_writer :organization_name
+
+    validate :organization_name_resolves?
+
+    private
+
+    def organization_name_resolves?
+      return if find_organization_name != 'Organization not found'
+
+      errors.add(:organization_name, 'Organization not found')
+    end
+
+    def find_organization_name
```

Comment:
- I'm a little torn on the need for validating organization info against the db. The front end should have access to all of the organization information needed to fill out the form, it could simply pass the information. I have been burned by similar assumptions before, so I understand why we'd want to make sure the id corresponds to a real record.
- I`|| 'Organization not found'` isn't necessary. `org&.name` is enough to either return the name or nil. Then you can nil check for adding to the errors array.

Reply:
That makes sense, I'll make it happen.

Reply:
I ended up keeping the validation and moving to the straight `org&.name` for the final return.